### PR TITLE
Add --disable_warmup flag.

### PR DESCRIPTION
**Pull request by doru1004**
# Add --disable_warmup flag

- This PR adds a new flag which disables warmup.

companion PR to be approved and merged is https://github.com/LLNL/RAJAPerf/pull/307

### CODE REVIEW

**Comment by doru1004** [comment]
ping

File: src/common/RunParams.cpp [hunk 1]
```diff
@@ -526,9 +529,13 @@ void RunParams::parseCommandLineOptions(int argc, char** argv)
...
-       if (input_state != BadInput) {
-         input_state = DryRun;
-       }
+      if (input_state != BadInput) {
+        input_state = DryRun;
+      }
+
+    } else if ( std::string(argv[i]) == std::string("--disable_warmup") ) {
+
+      disable_warmup = 1;
```
Comment (MrBurmark) [inline src/common/RunParams.cpp:538]:
```suggestion
      disable_warmup = true;
```

File: src/common/RunParams.cpp [hunk 2]
```diff
@@ -54,7 +54,8 @@ RunParams::RunParams(int argc, char** argv)
    npasses_combiner_input(),
    invalid_npasses_combiner_input(),
    outdir(),
-   outfile_prefix("RAJAPerf")
+   outfile_prefix("RAJAPerf"),
+   disable_warmup(0)
```
Comment (MrBurmark) [inline src/common/RunParams.cpp:58]:
```suggestion
   disable_warmup(false)
```

**Review by MrBurmark — CHANGES_REQUESTED** [review]
This seems fine. Do you have any specific reasons for wanting to skip the warmups?

**Comment by doru1004** [comment]
> This seems fine. Do you have any specific reasons for wanting to skip the warmups?

It makes overall testing faster and avoids the corner case when any error in the warmup stops the actual tests from running.

**Comment by rhornung67** [comment]
@doru1004 we understand your comment about errors in the warmup preventing actual kernels from running. We have some thoughts on how to improve that. 

Meanwhile, we need to pull the branch from your fork into our repo so all CI checks can run before merging. I'll do that after you're done making changes to your fork branch. @MrBurmark please let me know when you have no more changes to suggest.

**Review by MrBurmark — APPROVED** [review]

**Comment by doru1004** [comment]
> @doru1004 we understand your comment about errors in the warmup preventing actual kernels from running. We have some thoughts on how to improve that.
> 
> Meanwhile, we need to pull the branch from your fork into our repo so all CI checks can run before merging. I'll do that after you're done making changes to your fork branch. @MrBurmark please let me know when you have no more changes to suggest.

Thanks a lot. The branch is stable from my side. If you have any advice for me going forward as to what the best way to open these PRs is, please let me know. I may have a few more PRs coming in this area of testing.

**Comment by rhornung67** [comment]
@doru1004 it's a pain, but we have to manage PRs from folks outside of the LLNL GitHub org this way due to security policies for running GitLab CI on Livermore Computing machines.
